core/src/main/scala/spinal/core/Latch.scala

```diff
@@ -0,0 +1,23 @@
+package spinal.core
+
+object Latch {
+
+  def apply[T <: Data](dataType: HardType[T]): T = {
+    dataType.apply().addTag(noLatchCheck)
+  }
+}
+
+object LatchWhen {
+
+  def apply[T <: Data](next: T, cond: Bool): T = {
+    val latch = next.clone()
+      .setCompositeName(next, "latchWhen", true)
+      .addTag(noLatchCheck)
+      .asInstanceOf[T]
```

Comment:
```suggestion
  def apply[T <: Data](next: HardType[T], cond: Bool): T = {
    val latch = Latch(next).setCompositeName(next, "latchWhen", true)
```

Is it possible to re-use code like this? See https://github.com/SpinalHDL/SpinalHDL/blob/603bfcc3a234f092242e37721450f406fae066b5/core/src/main/scala/spinal/core/Reg.scala#L60

Reply:
I wasn't aware that would work. It seems it does so I'll take it